src/Tasks/Hash.cs

```diff
@@ -37,61 +35,102 @@ public class Hash : TaskExtension
         [Output]
         public string HashResult { get; set; }
 
+        private static readonly char ItemSeparatorCharacter = '\u2028';
+
+        private static readonly Encoding encoding = Encoding.UTF8;
+
+        private static readonly byte[] ItemSeparatorCharacterBytes = encoding.GetBytes(new char[] { ItemSeparatorCharacter });
+
+        private const int sha1BufferSize = 512;
+
+        private const int maxInputChunkLength = 256;
+
         /// <summary>
         /// Execute the task.
         /// </summary>
         public override bool Execute()
         {
-            if (ItemsToHash?.Length > 0)
+            if (ItemsToHash != null && ItemsToHash.Length > 0)
             {
                 using (var sha1 = SHA1.Create())
                 {
-                    var concatenatedItemStringSize = ComputeStringSize(ItemsToHash);
+                    // Buffer in which bytes of the strings are to be stored until their number reachs the limit size.
+                    // Once the limit is reached, the sha1.TransformBlock is be run on all the bytes of this buffer.
+                    byte[] sha1Buffer = System.Buffers.ArrayPool<byte>.Shared.Rent(sha1BufferSize);
 
-                    var hashStringSize = sha1.HashSize;
+                    byte[] bytesBuffer = System.Buffers.ArrayPool<byte>.Shared.Rent(encoding.GetMaxByteCount(maxInputChunkLength));
 
-                    using (var stringBuilder = new ReuseableStringBuilder(Math.Max(concatenatedItemStringSize, hashStringSize)))
+                    int bufferLength = 0;
+                    for (int i = 0; i < ItemsToHash.Length; i++)
                     {
-                        foreach (var item in ItemsToHash)
+                        string itemSpec = IgnoreCase ? ItemsToHash[i].ItemSpec.ToUpperInvariant() : ItemsToHash[i].ItemSpec;
+
+                        // Slice the itemSpec string into chunks of reasonable size and add them to sha1 buffer.
+                        int itemSpecLength = itemSpec.Length;
+                        int itemSpecPosition = 0;
+                        while (itemSpecLength > 0)
```

Comment:
nit:
```suggestion
                        for (int itemSpecPosition = 0; itemSpecPosition < itemSpec.Length; itemSpecPosition += maxInputChunkLength)
```

which lets you remove the `itemSpecPosition += ...` and `itemSpecLength += ...` below.